miditok/tokenizations/pertok.py

```diff
@@ -0,0 +1,573 @@
+"""PerTok tokenizer."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import numpy as np
+from symusic import Note, Pedal, PitchBend, Score, Tempo, TimeSignature, Track
+
+from miditok.classes import Event, TokenizerConfig, TokSequence
+from miditok.constants import DEFAULT_VELOCITY, MIDI_INSTRUMENTS, TIME_SIGNATURE
+from miditok.midi_tokenizer import MusicTokenizer
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+    from numpy.typing import NDArray
+    from symusic.core import TimeSignatureTickList
+
+
+class PerTok(MusicTokenizer):
+    r"""
+    PerTok: Performance Tokenizer.
+
+    Created by Lemonaide
+    https://www.lemonaide.ai/
+
+    Designed to capture the full spectrum of rhythmic values
+    (16ths, 32nds, various denominations of triplets/etc.)
+    in addition to velocity and microtiming performance characteristics.
+    It aims to achieve this while minimizing both vocabulary size and sequence length.
+
+    Notes are encoded by 2-5 tokens:
+
+    TimeShift
+    Pitch
+    Velocity (optional)
+    MicroTiming (optional)
+    Duration (optional)
```

Comment:
```suggestion
    * TimeShift
    * Pitch
    * Velocity (optional)
    * MicroTiming (optional)
    * Duration (optional)
```

That should hopefully solve the docs warning (making the CI fail)